pgmpy/models/DoubleMLRegressor.py

```diff
@@ -0,0 +1,188 @@
+"""flake8: noqa"""
+
+from typing import Any, List, Optional, Sequence, Tuple
+
+import numpy as np
+import pandas as pd
+from sklearn.base import BaseEstimator, RegressorMixin, clone
+from sklearn.utils.validation import check_is_fitted, check_X_y
+
+
+class DoubleMLRegressor(BaseEstimator, RegressorMixin):
+    def __init__(
+        self,
+        dag: Any,
+        estimator_g: Any,
+        estimator_m: Optional[Any] = None,
+        adjustment_set: Optional[Sequence[str]] = None,
+        treatment_col: Optional[str] = None,
+        n_folds: int = 5,
+        random_state: Optional[int] = None,
+    ):
+        self.dag = dag
+        self.estimator_g = estimator_g
+        self.estimator_m = estimator_m
+        self.adjustment_set = adjustment_set
+        self.treatment_col = treatment_col
+        self.n_folds = n_folds
+        self.random_state = random_state
+
+    def _more_tags(self):
+        return {"requires_y": True, "no_sparse_input": True}
+
+    # ---------- small helpers ----------
+    def _as_dataframe(self, X) -> pd.DataFrame:
```

Comment:
I am not sure if this method is valid. Given an array, we have no way to know which column corresponds to which variable in the model.

Reply:
what do you mean with "mipmapping" - that is a computer graphics term which I am not sure how it applies